src/elements/core/base-elements/open-close-escapable-element.ts

```diff
@@ -0,0 +1,57 @@
+import { isServer } from 'lit';
+
+import { SbbOpenCloseBaseElement } from './open-close-base-element.js';
+
+/**
+ * The registry is mapped as an array of WeakRef: it has been noticed that changing page
+ * nor the `disconnectedCallback` nor other `unload` hooks are triggered,
+ * so if the change is done with an opened dialog in the view, its instance is not removed.
+ * This way the reference should be removed from the array when the garbage collector acts.
+ */
+const overlayStack = new Array<WeakRef<SbbOpenCloseEscapableElement>>();
+const registry = new FinalizationRegistry((element: WeakRef<SbbOpenCloseEscapableElement>) => {
+  overlayStack.splice(
+    overlayStack.findIndex((e) => e === element),
+    1,
+  );
+});
+
+if (!isServer) {
+  window.addEventListener('keydown', (event: KeyboardEvent) => {
+    if (event.key === 'Escape' && !event.defaultPrevented) {
+      event.preventDefault();
+      overlayStack.at(-1)?.deref()?.close();
+    }
+  });
+}
+
+/**
+ * Class for overlay components which extends SbbOpenCloseBaseElement
+ * and adds the functionality to close the overlay pressing the `Escape` button.
+ */
+export abstract class SbbOpenCloseEscapableElement extends SbbOpenCloseBaseElement {
+  /** Opens the component. */
+  public open(): void {
+    this._cleanStack();
+    const componentWeakRef = new WeakRef(this);
+    overlayStack.push(componentWeakRef);
+    registry.register(this, componentWeakRef);
+  }
+
+  /** Closes the component. */
+  public close(): void {
+    if (overlayStack.length) {
+      overlayStack.pop();
+    }
+  }
+
+  private _cleanStack(): void {
+    overlayStack
+      .map((overlay: WeakRef<SbbOpenCloseEscapableElement>) => overlay.deref()?.id)
+      .forEach((overlayId: string | undefined, index: number) => {
+        if (overlayId && !document.getElementById(overlayId)) {
```

Comment:
Is it really necessary to use an Id lookup? This would break shadow DOM usage. Couldn't we just use the element which is hold in overlayStack?

Reply:
I added this code when i noticed that opening a dialog and changing to a different story maintains the opened element in the stack (since the close method is not called and adding hooks like disconnectedCallback on unload won't work); 
so technically if you do this several time, the stack would be a bit bloated, and there would be several elements 'on hold'

my idea was that when you open another dialog in the new page, the old element should not be in the dom anymore and it would be removed;
but as you mentioned and as mentioned in the pr description, it breaks shadowDOM and if the id is reused it won't work, so i would appreciate any other solution :)

Reply:
Apparently, using the 'hostDisconnected' works, so after some testing I reverted to a more basic implementation